src/rgw/rgw_op.h

```diff
@@ -354,6 +354,19 @@ class RGWGetObj : public RGWOp {
 
   int get_data_cb(bufferlist& bl, off_t ofs, off_t len);
 
+  int submit_l2_request(string dest, off_t obj_ofs, size_t len, off_t read_ofs, void* args, size_t (*write_cb)(void *, size_t, size_t, void *)){
+    return s->submit_http_req(dest, obj_ofs, len, read_ofs, args, write_cb);
+  }
+  int get_req_info(string dest, string& uri, string& auth_token) {
+    return s->get_req_info(dest, uri, auth_token);
+  }
+  bool get_cache_request() {
+    char *oid =(char *)s->info.env->get("HTTP_D3N");
```

Comment:
1. is there any reason for a cast here? 2. is it casting away const?

Reply:
fixed to  `const char* oid = s->info.env->get("HTTP_D3N");`

Reply:
cool